### PR TITLE
Add autoyast xml and yaml scheduler for bcache test job

### PR DESCRIPTION
**Pull request by b10n1k**
Create a new yaml scheduler and the xml for the autoyast.
The yaml scheduler adds verification for the partitioning
and of the AY profile.

- Related ticket: https://progress.opensuse.org/issues/49823
- Verification run: 
http://aquarius.suse.cz/tests/1956

### CODE REVIEW

File: data/autoyast_sle15/autoyast_bcache.xml [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        <version>{{VERSION}}</version>
+        <arch>{{ARCH}}</arch>
+      </addon>
+      <addon>
+        <name>sle-module-server-applications</name>
+        <version>{{VERSION}}</version>
+        <arch>{{ARCH}}</arch>
+      </addon>
+      <addon>
+        <name>sle-module-desktop-applications</name>
```
Comment (jknphy) [inline data/autoyast_sle15/autoyast_bcache.xml]: let's remove this module, no needed to test bcache and we save installation time.

File: data/autoyast_sle15/autoyast_bcache.xml [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      <global>
+          <timeout config:type="integer">-1</timeout>
+      </global>
+  </bootloader>
+  <general>
+    <mode>
+      <confirm config:type="boolean">false</confirm>
+    </mode>
+  </general>
+  <keyboard>
```
Comment (jknphy) [inline data/autoyast_sle15/autoyast_bcache.xml]: you can delete 'keyboard', 'language', 'ntp-client', 'networking', 'firewall', 'timezone' sections as they are default.

File: data/autoyast_sle15/autoyast_bcache.xml [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+        </partition>
+        <partition>
+          <mount>swap</mount>
+          <filesystem config:type="symbol">swap</filesystem>
+          <size>auto</size>
+        </partition>
+      </partitions>
+    </drive>
+  </partitioning>
+  <software>
```
Comment (jknphy) [inline data/autoyast_sle15/autoyast_bcache.xml]: this section 'software' you need to figure out what you can remove, perhaps you need some pattern, but many things will be default too. Check other minimal profiles, 'mini' 'multi-btrfs', recent ones.

File: data/autoyast_sle15/autoyast_bcache.xml [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+  </networking>
+  <firewall>
+    <enable_firewall config:type="boolean">true</enable_firewall>
+    <start_firewall config:type="boolean">true</start_firewall>
+  </firewall>
+  <timezone>
+    <hwclock>UTC</hwclock>
+    <timezone>Europe/Berlin</timezone>
+  </timezone>
+  <users config:type="list">
```
Comment (jknphy) [inline data/autoyast_sle15/autoyast_bcache.xml]: Check 'users' as well, sometimes is not needed as far as I remember, if you just need to use root for the validation. If so, in the validation below we can use another path not relative to `bernhard`.

File: tests/console/verify_bcache_partitioning.pm [hunk 5]
```diff
@@ -0,0 +1,57 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
```
Comment (jknphy) [inline tests/console/verify_bcache_partitioning.pm]: s/2019/2020/

File: tests/console/verify_bcache_partitioning.pm [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Validate bcache in writethrough
+# The backing device is /home and it is attached to the caching device which is the second
+# disk. Ususally the caching device is a fast device (ssd). In this scenario a rotational device is used.
```
Comment (jknphy) [inline tests/console/verify_bcache_partitioning.pm]: s/Ususally/Usually/

File: tests/console/verify_bcache_partitioning.pm [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+# This doesnt give any significant performance advantages but it is better to test bcache with this setup
+# than use bcache without caching(which yast provides a option if we want to do so)
+# Scenarios covered:
+# - Verify that certain values are set correctly in a bcache setup after installation;
```
Comment (jknphy) [inline tests/console/verify_bcache_partitioning.pm]: s/;/./

File: tests/console/verify_bcache_partitioning.pm [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+use scheduler 'get_test_suite_data';
+use Test::Assert 'assert_true';
+
+sub run {
+
+    my $test_data = get_test_suite_data();
+
+    select_console 'root-console';
+    # assert the registered backing dev
+    assert_script_run "cat /sys/block/bcache0/bcache/backing_dev_name | grep vda4";
```
Comment (jknphy) [inline tests/console/verify_bcache_partitioning.pm]: Let's use test_data to store `vda4` and `vdb`?

File: tests/console/verify_bcache_partitioning.pm [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    # assert bcache is running
+    assert_script_run "cat /sys/block/bcache0/bcache/running | grep 1";
+    # assert bcache is setup and is not cached dirty data
+    assert_script_run "cat /sys/block/bcache0/bcache/state | grep clean";
+
+    assert_script_run "lsblk";
```
Comment (jknphy) [inline tests/console/verify_bcache_partitioning.pm]: what about instead of just displaying partitioning, reuse module `validate_file_system` and schedule it before this one?

File: tests/console/verify_bcache_partitioning.pm [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+    record_info("bcache info", "Show info of bcache /dev/vdb");
+    assert_script_run "bcache-super-show /dev/vdb";
+    assert_script_run "cat /proc/partitions";
+    #validate_script_output "cat /proc/partitions", sub { m/\/tmp.*No_COW/i; };
+    record_info("write operation");
+    my $hits = script_output "cat /sys/block/bcache0/bcache/stats_total/cache_hits";
```
Comment (jknphy) [inline tests/console/verify_bcache_partitioning.pm]: more clear would be `hit_i` and `hit_f` which stand for initial and final.

File: schedule/yast/autoyast_bcache.yaml [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+test_data:
+  backingdev: vda4
+  profile:
+    partitioning:
+      - drive:
+          unique_key: device
+          device: /dev/vda
+          type: CT_DISK
+          partitions:
+            - partition:
```
Comment (jknphy) [inline schedule/yast/autoyast_bcache.yaml]: validator for cloned profile doesn't want to be smart if you just include a subset of partitions, AutoYaST generates the boot partition by default, so you are missing that one here so it will match 4.

File: data/autoyast_sle15/autoyast_bcache.xml (hunk 3 above)
Comment (b10n1k) [reply data/autoyast_sle15/autoyast_bcache.xml]: i think thats quite ok as it is.

File: data/autoyast_sle15/autoyast_bcache.xml (hunk 4 above)
Comment (b10n1k) [reply data/autoyast_sle15/autoyast_bcache.xml]: i would rather leave it as it is as it doesnt make any difference.

File: tests/console/verify_bcache_partitioning.pm (hunk 7 above)
Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: i am pretty sure that you can find this in other modules as well. And matter of taste here. Lets avoid this

File: data/autoyast_sle15/autoyast_bcache.xml (hunk 3 above)
Comment (jknphy) [reply data/autoyast_sle15/autoyast_bcache.xml]: it is not, we don't store cloned profile there, we store minimal AutoYaST profiles. I'm not sure if I can give you all the reason for that, but from the top of my head I believe the most important is that if something change in how AutoYaST generate those defaults some tags not related with the goal of the profile itself will not remains the same, so as less data we have less test maintenance, the other one is just that is more useful for us looking at other profiles to continue creating minimal ones and they are less polluted.
It is a little tricky and you have to go through the process several time to know what to pull out, but then you understand better the defaults.

File: tests/console/verify_bcache_partitioning.pm (hunk 7 above)
Comment (jknphy) [reply tests/console/verify_bcache_partitioning.pm]: a semicolon at the end of a line looks like a typo, though. IMHO a comment is a comment not a line in a programming language which ends line with semicolon. I guess that there are plenty of grammatical errors in the repo but no need for repeat them, I guess.

File: tests/console/verify_bcache_partitioning.pm (hunk 9 above)
Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: let me take a look on this.

File: schedule/yast/autoyast_bcache.yaml (hunk 11 above)
Comment (b10n1k) [reply schedule/yast/autoyast_bcache.yaml]: Dawm. thanks for it. I am stupid as i was looking to a wrong xml too.

Comment (jknphy) [reply schedule/yast/autoyast_bcache.yaml]: no worries, happy to help ;)

File: tests/console/verify_bcache_partitioning.pm (hunk 7 above)
Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: i will change it but this goes against the flexibility which quite confuse me. if there are not restrictions we should not review those kind of typos. I have fixed the rest of the grammatical errors but this is something else entirely.

File: data/autoyast_sle15/autoyast_bcache.xml (hunk 3 above)
Comment (b10n1k) [reply data/autoyast_sle15/autoyast_bcache.xml]: @jknphy i think you gave me a very good explanation but the software section is the same with other AY profiles. definetely i need to study those cases but i cant see serious reason to change them now. Mainly for the reason above which IMO means i would have to take care the others too.

File: data/autoyast_sle15/autoyast_bcache.xml (hunk 1 above)
Comment (b10n1k) [reply data/autoyast_sle15/autoyast_bcache.xml]: f35918e67

File: data/autoyast_sle15/autoyast_bcache.xml (hunk 2 above)
Comment (b10n1k) [reply data/autoyast_sle15/autoyast_bcache.xml]: i did removed all of them but it started breaking the installation. So i moved them all back. Again this is the minimal i see in other files too.

File: tests/console/verify_bcache_partitioning.pm (hunk 5 above)
Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: f35918e67

File: tests/console/verify_bcache_partitioning.pm (hunk 7 above)
Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: f35918e67

File: tests/console/verify_bcache_partitioning.pm (hunk 8 above)
Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: se test_data to st

Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: f35918e

File: tests/console/verify_bcache_partitioning.pm (hunk 9 above)
Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: f35918e

File: tests/console/verify_bcache_partitioning.pm (hunk 10 above)
Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: f35918e

**Comment by b10n1k** [comment]
@jknphy Do you know if the clone-autoinst.xml has generated correctly? I am talking about that extra partition on vda. The reason i am concerned is because you said that this should be the /boot partition but it has not mount node. I can check tmr but if you have any idea it would save some time

File: tests/console/verify_bcache_partitioning.pm (hunk 7 above)
Comment (jknphy) [reply tests/console/verify_bcache_partitioning.pm]: Flexibility is that you can do things differently, for example, use different kind of commands for your validation. You can share your view and have and open discussion giving your arguments and reasoning. 
For this particular case, grammatical errors are usually addressed in PRs. Of course if you insist on not changing it I would merge anyway as it is a minor thing but it is a public repo and it is for our own good to correct those mistakes.

File: data/autoyast_sle15/autoyast_bcache.xml (hunk 2 above)
Comment (jknphy) [reply data/autoyast_sle15/autoyast_bcache.xml]: that is expected, it is tricky to find the minimal one, other files that you could have checked are not really minimal. Check [mini profile](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/data/autoyast_sle15/mini.xml)
I think we should not pull out 'networking', the rest should be fine to delete them.

File: data/autoyast_sle15/autoyast_bcache.xml (hunk 3 above)
Comment (jknphy) [reply data/autoyast_sle15/autoyast_bcache.xml]: Check [mini profile](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/data/autoyast_sle15/mini.xml) for minimal software section, some packages and pattern are installed by default.

**Comment by jknphy** [comment]
> @jknphy Do you know if the clone-autoinst.xml has generated correctly? I am talking about that extra partition on vda. The reason i am concerned is because you said that this should be the /boot partition but it has not mount node. I can check tmr but if you have any idea it would save some time

this boot partition are not mounted afik: https://openqa.suse.de/tests/3927900#step/raid_gpt/20

File: schedule/yast/autoyast_bcache.yaml [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+  - autoyast/console
+  - autoyast/login
+  - autoyast/clone
+  - autoyast/repos
+  - autoyast/logs
+  - autoyast/autoyast_reboot
+  - installation/grub_test
+  - installation/first_boot
+  - console/verify_bcache_partitioning
+  - console/validate_file_system
```
Comment (jknphy) [inline schedule/yast/autoyast_bcache.yaml]: we could schedule `validate_file_system` before `verify_bcache_partitioning` as it is more generic, so if the file system is ok we go on.

File: tests/console/verify_bcache_partitioning.pm [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Validate bcache in writethrough
+# The backing device is /home and it is attached to the caching device which is the second
```
Comment (jknphy) [inline tests/console/verify_bcache_partitioning.pm]: it is better described in the ticket: `Bcache uses not formatted native linux partition as backing device and second disk as a whole as caching device.` in fact /home is not the backing device, is one of the mount point for the cached device. The backing device is in the 4th partition of vda.

**Comment by jknphy** [comment]
@b10n1k might I ask you why you didn't use the profile attached in the ticket or something similar? I don't think here make much more sense to have two SWAP partitions, one fixed and another one speed-up by this cache mechanism. In the profile provided in the ticket is `/` and `swap` what is cached, as using /home or not is optional.

File: tests/console/verify_bcache_partitioning.pm (hunk 9 above)
Comment (jknphy) [reply tests/console/verify_bcache_partitioning.pm]: that module is definitely work in progress, but at least I see that is good that we are reusing it somehow, would you mind to include in that module this line that you just delete it `assert_script_run "lsblk"` because now it is not visible [this](http://aquarius.suse.cz/tests/1910#step/verify_bcache_partitioning/14) which is really helpful, right?
Another thing is if you want to check the whole partitioning, not just the bcache part as everything is connected in this setup, but the module probably would not allow you to do that, so unless you see it easy to do now, just leave it as it is and we can improve in future.

File: data/autoyast_sle15/autoyast_bcache.xml (hunk 2 above)
Comment (b10n1k) [reply data/autoyast_sle15/autoyast_bcache.xml]: I removed some. you can check later VR

File: data/autoyast_sle15/autoyast_bcache.xml (hunk 3 above)
Comment (b10n1k) [reply data/autoyast_sle15/autoyast_bcache.xml]: I get it. What i am objecting to, is that the xml is minimal enough as it is. The packages that they are in the xml will be installed anyway. Anyway i reduced them.

File: schedule/yast/autoyast_bcache.yaml [hunk 14]
```diff
@@ -0,0 +1,81 @@
+---
+name: autoyast_bcache
+description: |
```
[the remark this answers is not in the document]
Comment (b10n1k) [reply schedule/yast/autoyast_bcache.yaml]: No this is wrong. The description is formatted in such way and the block style is used to keep the lines. If we do not care for the lines, i apologize.

File: schedule/yast/autoyast_bcache.yaml (hunk 12 above)
Comment (b10n1k) [reply schedule/yast/autoyast_bcache.yaml]: thats correct but for technical reasons i put that order. If I change validate_file_system to work here i have to VR the other jobs that they use it

File: tests/console/verify_bcache_partitioning.pm [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+# disk. Usually the caching device is a fast device (ssd). In this scenario a rotational device is used.
+# This doesnt give any significant performance advantages but it is better to test bcache with this setup
+# than use bcache without caching(which yast provides a option if we want to do so)
+# Scenarios covered:
+# - Verify that certain values are set correctly in a bcache setup after installation.
+# - Write operation in a backing device can be performed and validate that cache is working watching the cache hits.
+# Maintainer: yiannis bonatakis <ybonatakis@suse.com>
```
[the remark this answers is not in the document]
Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: I know. But i dont really care about it

File: schedule/yast/autoyast_bcache.yaml [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+  - autoyast/repos
+  - autoyast/logs
+  - autoyast/autoyast_reboot
+  - installation/grub_test
+  - installation/first_boot
+  - console/verify_bcache_partitioning
+  - console/validate_file_system
+  - autoyast/verify_cloned_profile
+test_data:
+  backingdev: vda4
```
[the remark this answers is not in the document]
Comment (b10n1k) [reply schedule/yast/autoyast_bcache.yaml]: Come on! what is this?

File: tests/console/verify_bcache_partitioning.pm (hunk 13 above)
Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: this looks to me like a description of the bcache rather of the test scenario

File: schedule/yast/autoyast_bcache.yaml (hunk 14 above)
[the remark this answers is not in the document]
Comment (jknphy) [reply schedule/yast/autoyast_bcache.yaml]: ahm I see, I've just check [yaml-multiline](https://yaml-multiline.info/) I didn't know that other possibility, probably doesn't make any difference. thanks for the clarification.

File: data/autoyast_sle15/autoyast_bcache.xml (hunk 3 above)
Comment (jknphy) [reply data/autoyast_sle15/autoyast_bcache.xml]: looks much more better now, thanks!

File: schedule/yast/autoyast_bcache.yaml (hunk 12 above)
Comment (jknphy) [reply schedule/yast/autoyast_bcache.yaml]: I'm not following you here, if you just switch the order or those two you would not affect any other test suite and as far as you have some root console you would not need to run the VR because would probably work. Could you clarify?

File: tests/console/verify_bcache_partitioning.pm (hunk 13 above)
Comment (jknphy) [reply tests/console/verify_bcache_partitioning.pm]: sure, it is a definition, but when you tried to be specific mentioned /home as a backing device, do you think that is correct? is the partition, right?

File: schedule/yast/autoyast_bcache.yaml (hunk 16 above)
[the remark this answers is not in the document]
Comment (jknphy) [reply schedule/yast/autoyast_bcache.yaml]: it is just about to use just one naming convention not mixing them, but let's discuss offline this point please if you have further comment.

File: data/autoyast_sle15/autoyast_bcache.xml (hunk 2 above)
Comment (jknphy) [reply data/autoyast_sle15/autoyast_bcache.xml]: Perfect, thanks!

File: schedule/yast/autoyast_bcache.yaml (hunk 12 above)
Comment (b10n1k) [reply schedule/yast/autoyast_bcache.yaml]: Maybe this will be more helpful http://aquarius.suse.cz/tests/1916#

File: tests/console/verify_bcache_partitioning.pm (hunk 13 above)
Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: i think it is correct. I try to describe what the job is intended to do. So /home is used as a backing device (/dev/vda) with the second disk as caching device (/dev/vdb) as probably would be in a really setup which uses bcache. i guess the description could be better so i am open for suggestions.

File: tests/console/verify_bcache_partitioning.pm [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+    assert_script_run "cat /proc/partitions";
+
+    record_info("write operation");
+    my $hit_i = script_output "cat /sys/block/bcache0/bcache/stats_total/cache_hits";
+    record_info "$hit_i";
+    my $path_in_backingdev = "/home/bernhard/one/two/three";
+    assert_script_run "mkdir -p $path_in_backingdev";
+    assert_script_run "touch ${path_in_backingdev}/data";
+    assert_script_run "dd bs=1024 count=1000000 < /dev/random > ${path_in_backingdev}/data";
+
```
Comment (JRivrain) [inline tests/console/verify_bcache_partitioning.pm]: Looks good, but I would suggest an additional checksum test here. So instead of some random data, write something fixed, then verify its checksum.

Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: will it test bcache? The dd is used only to cause some write cache operation. We can check cache-miss and others  examining the files sys/block/bcache0/bcache/ and/or bcache-super-show

File: schedule/yast/autoyast_bcache.yaml (hunk 12 above)
Comment (jknphy) [reply schedule/yast/autoyast_bcache.yaml]: To achieve that order, you need `select_console 'root-console'`, just rebase your code, I already added it in another PR.

File: tests/console/verify_bcache_partitioning.pm (hunk 13 above)
Comment (jknphy) [reply tests/console/verify_bcache_partitioning.pm]: the backing device is a specific partition in the first disk (/dev/vda3?), which /home is its mount point and regarding the cache device I'd rephrase it to "the whole second disk" as well (meaning no partitions).

Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: 2ffcdfc00

File: schedule/yast/autoyast_bcache.yaml (hunk 12 above)
Comment (b10n1k) [reply schedule/yast/autoyast_bcache.yaml]: order is resolved

File: tests/console/verify_bcache_partitioning.pm (hunk 9 above)
Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: added

Comment (jknphy) [reply tests/console/verify_bcache_partitioning.pm]: I was referring to add it in `console/validate_file_system` but it is ok.

File: tests/console/verify_bcache_partitioning.pm (hunk 17 above)
Comment (jknphy) [reply tests/console/verify_bcache_partitioning.pm]: I guess to exercise the tooling for bcache here should suffice. Now that we are using / partition in bcache I guess we are already testing implicitly.

File: schedule/yast/autoyast_bcache.yaml [hunk 18]
```diff
@@ -0,0 +1,76 @@
+---
+name: autoyast_bcache
+description: |
+  Test installation with AY profile for bcache with encrypted lvm.
```
Comment (jknphy) [inline schedule/yast/autoyast_bcache.yaml]: is there any encrypted lvm?

File: tests/console/verify_bcache_partitioning.pm [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use scheduler qw(get_test_suite_data get_data_value_from);
+use Test::Assert 'assert_true';
+
+sub run {
+    my $test_data  = get_test_suite_data();
+    my $cachingdev = get_data_value_from($test_data->{profile}->{partitioning}->[1]->{drive}->{device});
```
Comment (jknphy) [inline tests/console/verify_bcache_partitioning.pm]: I wonder myself if we should use test data for each module that we validate, the problem to reuse it here is that the expression is pretty long and the order could change, to use indexes is always a bad idea :) I would create separate key for this data so it would be more readable. But leave it as it is if you prefer.

File: lib/scheduler.pm [hunk 20]
```diff
@@ -70,6 +70,19 @@ sub get_test_suite_data {
...
 }
 
+
+=head2 get_data_value_from
+
+Returns specific value of test_suite_data.
+
+=cut
+
+sub get_data_value_from {
```
[the remark this answers is not in the document]
Comment (jknphy) [reply lib/scheduler.pm]: I believe you didn't want to return the same value that you passed and call the function instead.

File: tests/console/verify_bcache_partitioning.pm [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+    select_console 'root-console';
+    my $partitioning = script_output "lsblk";
+    record_info "lsblk", "$partitioning";
```
Comment (jknphy) [inline tests/console/verify_bcache_partitioning.pm]: I would have preferred to have this `validate_file_system` instead of here, but feel free to leave it as it is.

File: schedule/yast/autoyast_bcache.yaml (hunk 18 above)
Comment (b10n1k) [reply schedule/yast/autoyast_bcache.yaml]: nope. removed

File: lib/scheduler.pm (hunk 20 above)
[the remark this answers is not in the document]
Comment (b10n1k) [reply lib/scheduler.pm]: know that i am thinking this is not doing what was indented to. I will remove it. the idea was to make a small step to avoid to have duplicated test data. Anyway. ignore

File: tests/console/verify_bcache_partitioning.pm (hunk 19 above)
Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: see above

File: tests/console/verify_bcache_partitioning.pm (hunk 21 above)
Comment (b10n1k) [reply tests/console/verify_bcache_partitioning.pm]: no problem. 9c7565875

Comment (jknphy) [reply tests/console/verify_bcache_partitioning.pm]: it is less (for me! :) ) clear `lsblk -l` than `lsblk`, the tree structure helps, but it is just a matter of taste, thanks for adding it.
